### PR TITLE
[ SERVICE-124] - Integrate tab UI with Snap & Dock service

### PR DESCRIPTION
**Pull request by JohnathonLamOF**
Integration of tabbing into develop layouts service.

Use the bottom row of buttons added to the layout manager UI to create some tabbable windows.

**Known issues:**
- The windows that are part of the tab demo are initialised with a tab strip, this is against spec (strip should only appear when there's two or more windows tabbed-together). There's a separate story to change this behaviour.
- Currently no way to tab-together windows that aren't initialised with a tab bar (separate story for this).
- Tabbing doesn't play nicely with snap & dock - if testing this PR, best to close the S&D demo windows before opening a tabbed window. There's a separate story to properly integrate tabbing and S&D.
- Demo app needs some work (due to it being three separate demos mashed together), will need to define and address separately.

### CODE REVIEW

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@johnathonlamof). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@johnathonlamof). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (MichaelMCoates) [inline .prettierrc]: Why did we add Prettier when we already have a linter?

File: package.json [hunk 2]
```diff
@@ -34,8 +35,10 @@
     "extract-text-webpack-plugin": "^3.0.2",
     "gts": "^0.7.1",
     "hadouken-js-adapter": "^0.32.1-alpha.5",
+    "jest": "^23.5.0",
```
Comment (MichaelMCoates) [inline package.json]: We've got jest, mocha, and ava all in one repo, so we gotta consolidate that, but we can do it later on.

File: .prettierrc (hunk 1 above)
Comment (MichaelMCoates) [reply .prettierrc]: 👍

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@johnathonlamof). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community! We require contributors to sign a [Contributor License Agreement](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/75530375/Legal+Requirements#LegalRequirements-ContributorLicenseAgreement) and we don't seem to have CLAs on file for these contributors to the Pull Request: (@johnathonlamof). In order for your PR to be reviewed and merged, please follow the directions at the link above.

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If there are any questions, please don't hesitate to [get in touch with our Infrastructure Support team](https://finosfoundation.atlassian.net/wiki/spaces/FINOS/pages/77955298/Getting+in+Touch#GettinginTouch-InfrastructureSupport). /CC @finos-admin

File: package.json (hunk 2 above)
Comment (pjbroadbent) [reply package.json]: Agreed. But we'll first need to pick which framework we want to go with. Enough to discuss/port-over that we should approach separately.

**Pull request merged**